examples/vnet/doca_dpdk.c

```diff
@@ -740,25 +787,53 @@ static void doca_dpdk_build_raw_data(uint8_t **header,
 
 	encap->conf.data = encap->data;
 	encap->conf.size = header - encap->data;
-	action->conf = &encap->conf;
+
+	if (hw_steering) {
+		action->type = RTE_FLOW_ACTION_TYPE_RAW_ENCAP;
+		action->conf = &encap->conf;
+	} else {
+		id = doca_encap_table_add_id(encap_data);
+		meta = &entry->action_data.meta;
+		if (doca_encap_table_get_refcnt(id) == 1) {
+			if (!doca_dpdk_add_tx_encap(pipe->port_id, id, &encap->conf)) {
+				meta->data = id;
+				meta->conf.data = meta->data;
+				meta->conf.mask = UINT32_MAX;
+			}
```

Comment:
what you do on fail here?
better have a least a log, and make sure you don't crash,

Reply:
Done.

Reply:
fail on add_tx_encap? in that case the meta will not be updated. I don't see how we can crash